### PR TITLE
Bump gulp-debian library

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: https://github.com/betaflight/betaflight-configurator/issues/3154

The `gulp-debian` library has been updated with `xz` as default compression type..

### CODE REVIEW

**Comment by haslinghuis** [comment]
@nerdCopter please check

**Comment by haslinghuis** [comment]
@McGiverGim build size stays the same. Why is it much larger then other `archs`?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by McGiverGim — CHANGES_REQUESTED** [review]
I think this PR is wrong. The `compressFiles` is only used to generate the portable version of the Configurator. For the `.deb´ version all is in the `release_deb` method.
I think remember we have a complete discussion about this in an earlier issue?

**Comment by McGiverGim** [comment]
Here was the discussion: https://github.com/betaflight/betaflight-configurator/pull/2930

**Comment by nerdCopter** [comment]
[Betaflight-Configurator-Linux](https://github.com/betaflight/betaflight-configurator/suites/9944880673/artifacts/481568230)
 
 `dpkg-deb: error: archive 'betaflight-configurator_10.9.0-debug-dff477f_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up`
 
`gulp-debian` maintainer is not responsive. research alludes that he left coding outside of employment.  I've personally tried a few (failing) methods to get this to work, but i only thing modding gulp-debian will be the fix.

**Comment by nerdCopter** [comment]
One method would be `dpk-deb -Zxz ...` on this line: https://github.com/stpettersens/gulp-debian/blob/189c808db3d146fcbcaed62182a177bd58aa1b2a/index.js#L194

Another should be: but i failed at all attempts
image
via https://raphaelhertzog.com/2010/09/17/how-to-create-debian-packages-with-alternative-compression-methods/

File: .github/workflows/ci.yml
Comment (nerdCopter) [inline .github/workflows/ci.yml:88]: `          if: ${{ matrix.name == 'Linux' }}`

Comment (haslinghuis) [reply .github/workflows/ci.yml:88]: Yes was waiting for the workflow to continue.

Did you check compression?

Comment (nerdCopter) [reply .github/workflows/ci.yml:88]: last one failed; waiting for latest.

**Comment by nerdCopter** [comment]
`4e30b3d` still failed :(
`dpkg-deb: error: archive 'betaflight-configurator_10.9.0-debug-4e30b3d_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up`

**Comment by haslinghuis** [comment]
Guess the alias does not transfer between different run items. In last push the alias should now be available.

**Comment by nerdCopter** [comment]
[Betaflight-Configurator-Linux](https://github.com/betaflight/betaflight-configurator/suites/9952572522/artifacts/482106205)
`c1beb0f`  fail, same.

**Comment by nerdCopter** [comment]
@haslinghuis , i may have a less obtrusive solution (repack .deb).  working on it now.

**Comment by nerdCopter** [comment]
`3f912d6` failed.

**Comment by McGiverGim** [comment]
> @McGiverGim might have found something useful for packaging using github actions workflow: https://github.com/marketplace/actions/build-deb-action. Maybe you have time to investigate.

I don't think this is a good option. We are generating the packages in the gulp file because they are available to the developer. Doing all of them in the gulpfile but the .deb in the action breaks the consistence, I think we need to go this way only as a latest option. And I think it will not work neither, the base image is ubuntu, the same than we use, so it will generate the .deb in the same way the gulp does, using calls to the package manager of the OS.

**Comment by nerdCopter** [comment]
* fwiw, i tried an alias locally within the same bash session resulting as failure as well. (still used zst)
* i'm surprised #3157 was merged so quickly, i was open and interested in alternative solutions.

**Comment by haslinghuis** [comment]
Please check the resulting build.

- Made the library locally.
- Made changes to remove security issues and sonar issues.
- Added the parameter to the call  `dpkg-deb -Zgzip --build`

**Comment by nerdCopter** [comment]
[Betaflight-Configurator-Linux](https://github.com/betaflight/betaflight-configurator/suites/9980405317/artifacts/484097671)  Does not contain .deb file

**Comment by haslinghuis** [comment]
I should revisit the change to the regex then :)

**Comment by McGiverGim** [comment]
I don't like the idea of having gulp-debian in the repo. If you can make it working, why don't you publish it in the npm repo as a fork? In this way we can use it here in a cleaner way.

**Comment by haslinghuis** [comment]
Ok, made a NPM account. Made a detach fork ticket in github.

As this is an MIT license it this even allowed?

**Comment by McGiverGim** [comment]
I don't know too much about licenses, but I have found this:
>Modify the code: In other words, developers can change/update the code however they'd like. Distribute copies of the code and any modifications: As long as the original copyright notice and the license itself are included, an organization can distribute and sell copies or modified versions of the code

**Comment by haslinghuis** [comment]
Good news:

https://github.com/stpettersens/gulp-debian/pull/26 is merged.

**Comment by nerdCopter** [comment]
> Good news:
> 
> [stpettersens/gulp-debian#26](https://github.com/stpettersens/gulp-debian/pull/26) is merged.

awesome!!  will have to revert #3157

**Comment by McGiverGim** [comment]
gulp-debian 0.2.0 released 😁

**Review by McGiverGim — APPROVED** [review]

**Comment by haslinghuis** [comment]
Some other merge in the library seem to have caused a problem. A fix is already submitted. So we need to wait for 2.0.1 :)

**Comment by haslinghuis** [comment]
We are now on version 0.3.2 :)

**Review by McGiverGim — APPROVED** [review]
Let's go for it! 😋

**Review by KarateBrot — APPROVED** [review]

**Comment by nerdCopter** [comment]
`04357b0` tested as installable on Debian 11 and Xubuntu 22.04 :muscle:

File: gulpfile.js
Comment (haslinghuis) [inline gulpfile.js:14]: This line can be removed

**Pull request merged**
